internal/caller/linux/linuxconfig.go

```diff
@@ -0,0 +1,136 @@
+package caller
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+var (
+	installPath   = "/etc/wireguard"
+	tunnelTemplate = `[Interface]
+PrivateKey = %s
+Address = %s
+	
+[Peer]
+PublicKey = %s
+AllowedIPs %s
+Endpoint = %s`
+)
+
+// LinuxConfig represents the tunnel configuration for a linux machine.
+type LinuxConfig struct {
+	localDevice wgtypes.Device
+	tunnel      string
+	confName    string
+	allowedIps  string
+	addr        string
+	endpoint    string
+}
+
+// init transmits the linux config factory to the caller Client.
+func init() {
+	AppConfigurator = NewConfig()
+}
+
+// NewConfig generates and returns a new LinuxConfig.
+func NewConfig() *LinuxConfig {
+	newDevice := wgtypes.Device{
+		Name: "Linux Configurator",
```

Comment:
This will try to create or find a device called "Linux Configurator".  While I think this would probably work, its conventional on linux for the adapters to be all lower case, have no spaces, and be followed by a number (think `eth0`).   This should probably be piped in as a variable by some means.